plugins/providers/gcloudiam/provider.go

```diff
@@ -99,11 +150,20 @@ func (p *Provider) RevokeAccess(pc *domain.ProviderConfig, a *domain.Appeal) err
 	}
 
 	if a.Resource.Type == ResourceTypeProject || a.Resource.Type == ResourceTypeOrganization {
-		if err := client.RevokeAccess(a.AccountType, a.AccountID, a.Role); err != nil {
-			if errors.Is(err, ErrPermissionNotFound) {
-				return nil
+		for _, rc := range pc.Resources {
+			for _, ro := range rc.Roles {
+				if a.Role == ro.ID {
+					for _, p := range ro.Permissions {
+						permission := fmt.Sprint(p)
+						if err := client.RevokeAccess(a.AccountType, a.AccountID, permission); err != nil {
+							if !errors.Is(err, ErrPermissionNotFound) {
+								return err
+							}
+						}
+					}
+					break
+				}
```

Comment:
Can you refactor this block too? this is highly nested and difficult to understand.

Reply:
Same as above